apps/frontend/components/BaseModal.tsx

```diff
@@ -0,0 +1,68 @@
+import { Loader2 } from 'lucide-react'
+import React, { Fragment, type ReactNode } from 'react'
+import {
+  AlertDialog,
+  AlertDialogContent,
+  AlertDialogDescription,
+  AlertDialogHeader,
+  AlertDialogOverlay,
+  AlertDialogTitle
+} from './shadcn/alert-dialog'
+
+interface BaseModalProps {
+  open: boolean
+  handleClose: () => void
+  children?: ReactNode
+  loading?: boolean
+  loadingMessage?: string
+  title?: string
+  description?: string
+  modalBgDarkMode?: boolean
+}
+
+/**
+ *
+ * @remarks
+ * * Use BaseModal Component by creating a new component(which includes 'AlertDialogFooter') that extends BaseModal
+ * * AlertDialogFooter section (Button section) is separated using ConfirmModal component for reusability.
+ */
+export default function BaseModal({
+  open,
+  handleClose,
+  children,
+  loading = false,
+  loadingMessage = '',
+  title = '',
+  description = '',
+  modalBgDarkMode = false
```

Comment:
```suggestion
  backgroundDarkMode = false
```

다 너무 좋은데, 사소한 의견을 더 하자면 `modalBgDarkMode` 대신 `backgroundDarkMode` 혹은 `darkMode`가 어떨까요? modal을 지우고 약어 대신 풀네임으로 쓰면 가독성이 더 좋아질 것 같아서요!